chaco/plots/scatterplot_1d.py

```diff
@@ -32,7 +32,9 @@ class ScatterPlot1D(Base1DPlot):
     marker = MarkerTrait
 
     # The pixel size of the marker, not including the thickness of the outline.
-    marker_size = Float(4.0)
+    marker_size = Union(Float,
+                        Array(dtype=float),
+                        default_value=4.0)
```

Comment:
This is absolutely not correct - Chaco data never comes from hard-coded arrays like this.  Most likely this should be a Float or an `Instance(AbstractDataSource)`.

Also make sure that you keep the default value of 4.0 otherwise lots of plots will break.